lib/Modules/ComputeHooks/ComputeHooks.psm1

```diff
@@ -445,61 +406,139 @@ function Get-CharmServices {
             )
         }
     }
-
+    $s2dContextGen = @{
+        "generator" = (Get-Item "function:Get-S2DCSVContext").ScriptBlock
+        "relation" = "csv"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'csv')) {
+        $s2dContextGen['mandatory'] = $true
+    }
+    $freeRdpContextGen = @{
+        "generator" = (Get-Item "function:Get-FreeRDPContext").ScriptBlock
+        "relation" = "free-rdp"
+        "mandatory" = $false
+    }
+    if((Confirm-JujuRelationCreated -Relation 'free-rdp')) {
+        $freeRdpContextGen['mandatory'] = $true
```

Comment:
Yes. But the deployer won't have any ideea if his relation with S2D was correctly made or not, right ? (if he adds the S2D relation, that is not ready for some whatever reason and the units keeps on being active)